consensus.go

```diff
@@ -313,6 +344,15 @@ func parseRouterFlags(flags []string) *RouterFlags {
 	return routerFlags
 }
 
+func parseIPv6AddressAndPort(addressAndPort string) (address net.IP, port uint16) {
+	var ipV6regex = regexp.MustCompile(`\[(.*?)\]`)
+	var ipV6portRegex = regexp.MustCompile(`\]:(.*)`)
+	address = net.ParseIP(ipV6regex.FindStringSubmatch(addressAndPort)[1])
+	port = StringToPort(ipV6portRegex.FindStringSubmatch(addressAndPort)[1])
+
+	return address, port
+}
```

Comment:
Using a regex may not be the quickest way here. For my purposes it is OK but I could look into speeding this up in the future.